powerauth-push-server/src/main/java/io/getlime/push/repository/model/PushMessageEntity.java

```diff
@@ -160,4 +160,24 @@ public Status getStatus() {
     public void setStatus(Status status) {
         this.status = status;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        PushMessageEntity that = (PushMessageEntity) o;
+
+        if (!getDeviceId().equals(that.getDeviceId())) return false;
+        if (!getMessageBody().equals(that.getMessageBody())) return false;
+        return getTimestampCreated().equals(that.getTimestampCreated());
+    }
+
+    @Override
+    public int hashCode() {
+        int result = getDeviceId().hashCode();
+        result = 31 * result + getMessageBody().hashCode();
+        result = 31 * result + getTimestampCreated().hashCode();
```

Comment:
There's a small bug on this row - contents of result are overwritten.

Reply:
I can't see what exactly is going to be overwritten, true is that attribute values should be used instead of getters.

Reply:
Sorry, I overlooked it, it is OK.